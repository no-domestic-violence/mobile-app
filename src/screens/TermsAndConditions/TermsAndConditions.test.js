import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import TermsAndConditionsScreen from './index';

jest.mock('react-i18next');
const mockNavigation = {
  navigate: jest.fn()
}
describe('<TermsAndConditionsScreen />', () => {
  it('should match snapshot', () => {
    const result = render(<TermsAndConditionsScreen />).toJSON();
    expect(result).toMatchSnapshot();
  });
  test('button should be not tappable by default', () => {
    const onPress = jest.fn();
    const tree = render(<TermsAndConditionsScreen />);
    const button = tree.getByTestId('terms-conditions-button');
    fireEvent.press(button);
    expect(onPress).not.toHaveBeenCalled();
  });
  test('button should be tappable when scrolled down near to the bottom', () => {
    const eventData = {
      nativeEvent: {
        layoutMeasurement: {
          height: 0
        },
        contentOffset: {
          y: 200
        },
        contentSize: {
          height: 80
        }
      }
    }
    const { getByTestId } = render(<TermsAndConditionsScreen navigation={mockNavigation}/>);
    fireEvent.scroll(getByTestId('terms-conditions-scroll-view'), eventData);
    fireEvent.press(getByTestId('terms-conditions-button'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('Onboarding');
  })
});
