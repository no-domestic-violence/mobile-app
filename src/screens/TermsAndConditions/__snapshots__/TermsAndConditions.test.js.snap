// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TermsAndConditionsScreen /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#cadeee",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <svg
    data-file-name="SvgLanguage"
    style={
      Object {
        "position": "absolute",
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 21,
        "marginTop": 40,
      }
    }
  >
    TermsAndConditions.title
  </Text>
  <RCTScrollView
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.71)",
        "borderRadius": 31,
        "height": 933.8,
        "marginTop": 30,
        "width": 637.5,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          Placeholder terms and conditions. If you continue to browse and use this website, you are agreeing to comply with and be bound by the following terms and conditions of use, which together with our privacy policy govern [business name]’s relationship with you in relation to this website. If you disagree with any part of these terms and conditions, please do not use our website.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          The term ‘[business name]’ or ‘us’ or ‘we’ refers to the owner of the website whose registered office is [address]. Our company registration number is [company registration number and place of registration]. The term ‘you’ refers to the user or viewer of our website.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           The content of the pages of this website is for your general information and use only. It is subject to change without notice.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           This website uses cookies to monitor browsing preferences. If you do allow cookies to be used, the following personal information may be stored by us for use by third parties: [insert list of information].
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           Neither we nor any third parties provide any warranty or guarantee as to the accuracy, timeliness, performance, completeness or suitability of the information and materials found or offered on this website for any particular purpose. You acknowledge that such information and materials may contain inaccuracies or errors and we expressly exclude liability for any such inaccuracies or errors to the fullest extent permitted by law.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           Your use of any information or materials on this website is entirely at your own risk, for which we shall not be liable. It shall be your own responsibility to ensure that any products, services or information available through this website meet your specific requirements.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           This website contains material which is owned by or licensed to us. This material includes, but is not limited to, the design, layout, look, appearance and graphics. Reproduction is prohibited other than in accordance with the copyright notice, which forms part of these terms and conditions.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           All trademarks reproduced in this website, which are not the property of, or licensed to the operator, are acknowledged on the website. Unauthorised use of this website may give rise to a claim for damages and/or be a criminal offence.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           From time to time, this website may also include links to other websites. These links are provided for your convenience to provide further information. They do not signify that we endorse the website(s). We have no responsibility for the content of the linked website(s).
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            }
          }
        >
          •
           Your use of this website and any dispute arising out of such use of the website is subject to the laws of England, Northern Ireland, Scotland and Wales.
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          The use of this website is subject to the following terms of use
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#999",
        "borderRadius": 20,
        "fontWeight": "600",
        "marginBottom": 40,
        "marginTop": 40,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
    testID="terms-conditions-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "600",
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      TermsAndConditions.button
    </Text>
  </View>
</View>
`;
